### ISSUE TITLE
fastly_tls_subscription return one challenge

### ISSUE THREAD

**Issue by hosht** [fastly/terraform-provider-fastly#430]
### Terraform Version
Terraform v1.0.0
on darwin_amd64
+ provider registry.terraform.io/fastly/fastly v0.32.0

### Affected Resource(s)
- fastly_tls_subscription

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
terraform {
  required_version = "1.0.0"

  required_providers {
    fastly = {
      source  = "fastly/fastly"
      version = "0.32.0"
    }
  }
}

locals {
  tls_domains = [
    "a.example.com",
    "b.example.com"
  ]
}

resource "fastly_tls_subscription" "this" {
  domains               = local.tls_domains
  certificate_authority = "lets-encrypt"
  configuration_id      = "configuration_id"
}
```

### Expected Behavior
when domain argument receives more than two domains, fastly_tls_subscription should return list of managed_dns_challenges and managed_http_challenges for domain validation.

```
"managed_dns_challenges": [
  {
    "record_name": "_acme-challenge.a.example.com",
    "record_type": "CNAME",
    "record_value": "abcde12345.fastly-validations.com"
  },
  {
    "record_name": "_acme-challenge.b.example.com",
    "record_type": "CNAME",
    "record_value": "abcde12345.fastly-validations.com"
  }
]
```

### Actual Behavior
it returns just a.example.com challenges. is it right?

```
"managed_dns_challenge": {
  "record_name": "_acme-challenge.a.example.com",
  "record_type": "CNAME",
  "record_value": "abcde12345.fastly-validations.com"
},
"managed_http_challenges": [
  {
    "record_name": "a.example.com",
    "record_type": "A",
...
```

**Comment by smaeda-ks** [comment]
Thanks, @hosht! This appears to be a bug in our provider.

As this is going to be a breaking change, we will be discussing the fix in https://github.com/fastly/terraform-provider-fastly/pull/435

**Comment by smaeda-ks** [comment]
This has been fixed in v0.33.0:
https://github.com/fastly/terraform-provider-fastly/blob/main/CHANGELOG.md

You can use the new `managed_dns_challenges` attribute instead:
https://registry.terraform.io/providers/fastly/fastly/latest/docs/resources/tls_subscription

We appreciate your feedback.

**Comment by hosht** [comment]
Thank you so much, it saves my time !
